Requirements.md

```diff
@@ -0,0 +1,101 @@
+# Requirements for GC
+
+This document identifies the key requirements for the managed memory proposal. Below we refer to the version of WASM with managed memory as being WASM-GC.
+
+## Goals
+
+1. Support for host-provided GC
+
+   Allow language implementers to use host-provided GC features as their primary means of supporting memory management. 
+   
+   On the whole, it would be preferable if language implementers could target WASM-GC as a natural target for their language. While it is difficult to be precise about this, an ideal scenario would be that language implementers could target WASM-GC as though it were just another back-end. In particular, targeting WASM-GC should try to avoid requiring changes to earlier stages of a compiler - such as the type checker or source-level transformations.
+   
+### Non-Goals
+1. Support for linear memory GC.
+
+   We do not attempt with this proposal to support alternate memory management strategies that rely on collecting linear memory.
+   
+## Criteria
+
+A criterion is intended to act as a metric of a 'soft value', such as 'desireability' or 'implementability'. Criteria are especially important when weighing the relative merits of features and in determining the importance that some scenario is satisfied.
+
+1. Prioriterize popular languages.
```

Comment:
> In particular, I'm sure we do not want to end up producing yet another OO-specific engine

The Lumen team is on the CG for this reason.  We wanted to make sure that Erlang/Elixir and more generally, functional languages would have good support in WASM.

Reply:
@rossberg 

>In particular, I'm sure we do not want to end up producing yet another OO-specific engine,

While I would personally be quite happy to see OO languages disappear, that is not going to happen; in part because it is not in our control. In fact, new OO languages are still being invented...

OTOH, that does not mean that WASM(-GC) should/must end up as an OO-specific engine. It simply means that WASM(-GC) must be seen as a friendly target for OO language compiler authors.

Reply:
@fgmccabe
>  It simply means that WASM(-GC) must be seen as a friendly target for OO language compiler authors.

No disagreement there. However, I get the rather concerning impression that many of the recent discussions reinterpret "being friendly" as "specialising for". It has to be equally friendly to others.